### DISCUSSION TITLE
Proper way of dealing with decorators?

### DISCUSSION THREAD

**Discussion by joezappie** [eslint/eslint#17795] [Q&A] [answered]
So whats the correct setup for dealing with decorators. I'm using legacy decorators. Before I had added:

```
  "parserOptions": {
    "ecmaFeatures": {
      "legacyDecorators": true
    }
  },
  "plugins": [
    "decorators-legacy"
  ],
```

Checking the `decorator-legacy` npm module is now deprecated with the tag: `Does not work, do not bother using` and wont install anymore:
https://www.npmjs.com/package/eslint-plugin-legacy-decorators

If I remove the plugin, the latest version of eslint still errors saying I need to enable the decorator-legacy plugin:
```
This experimental syntax requires enabling one of the following parser plugin(s): "decorators", "decorators-legacy".
```

But that plugin is deprecated! So what's the correct way to get eslint to not complain about decorators?

**Comment by Rec0iL99** [comment, marked as answer]
Hi @jrj2211, this is coming from [`@babel/eslint-parser`](https://github.com/babel/babel/discussions) which is a parser not maintained by the ESLint team. Please consult the maintainers there.

**Reply by joezappie** [reply]
Gotcha, thank you

**Comment by joezappie** [comment]
Just incase anyone else comes across this, the solution was to use the babel decorator plugin and pass in the plugins under parserOptions.babelOptions:

```
   // .eslintrc.json
  "parser": "@babel/eslint-parser",
  "parserOptions": {
    "ecmaVersion": 12,
    "sourceType": "module",
    "requireConfigFile": false,
    "babelOptions": {
      "plugins": [
        [
          "@babel/plugin-proposal-decorators",
          {
            "version": "2018-09",
            "decoratorsBeforeExport": true
          }
        ]
      ]
    }
  },
```

This optionally could also be done in a separate babelrc config, or in your package.json. I opted to put it in my `eslintrc.json` as I share my config as a package to use on all my projects.
